src/irmin-pack/unix/errors_base.ml

```diff
@@ -0,0 +1,67 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+(* TODO: Remove [`Tmp] everywhere *)
+
+type error =
+  [ `Double_close
+  | `File_exists
+  | `Invalid_parent_directory
+  | `No_such_file_or_directory
+  | `Not_a_file
+  | `Read_on_closed
+  | `Read_out_of_bounds
+  | `Ro_not_allowed
+  | `Tmp
+  | `Write_on_closed
+  | `Invalid_argument
+  | `Decoding_error
+  | `Not_a_directory
+  | `Invalid_layout
+  | `Corrupted_legacy_file
+  | `Pending_flush
+  | `Rw_not_allowed ]
+[@@deriving irmin ~pp]
+
+(* This exception does not contain [`Ro_not_allowed].
+
+   We can't use polyval inclusion because repr doesn't support it *)
+type error' =
```

Comment:
I don't understand why having `error'` (the patch compiles https://github.com/icristescu/irmin/commit/989ad1cd7b899bbbed596742ddbe3efe8ce951b0 but I maybe missed something). 

Also, why is Io_mis not included?

Reply:
Ok, thanks!